### PR TITLE
Update layout md plots

### PR DESCRIPTION
**Pull request by nathancoliver**
Below is a general list of changes:

- Defines plot format, including plot background color, and grid lines and border line color and line width.
- Automatically sets y-axis range. User currently does not have the ability to set the y-axis range.
- User can define plot title, and group plot lines into a legend group and provide a legend group name.
- Added units tests for legend_group and is_visible.

### CODE REVIEW

File: dimes/common.py [hunk 1]
```diff
@@ -253,6 +289,8 @@ def finalize_plot(self):
                                             "width": display_data.line_properties.marker_line_width,
                                         },
                                     },
+                                    legendgroup=display_data.legend_properties.legend_group,
+                                    legendgrouptitle={"text":display_data.legend_properties.legend_group},
```
Comment (nealkruis) [inline dimes/common.py:293]: This is a great addition!

**Review by nealkruis — CHANGES_REQUESTED** [review]
@nathancoliver, these are all great changes! I have a few comments on the implementation that I'm happy to go over with you on a call.

File: dimes/common.py [hunk 2]
```diff
@@ -182,6 +187,12 @@ def __init__(
             self.x_axis = x_axis
         self.subplots: List[Union[DimensionalSubplot, None]] = [None]
         self.is_finalized = False
+        self.GRID_LINE_WIDTH = 1.5
+        self.WHITE = "white"
+        self.BLACK = "black"
+        self.GREY = "rgba(128,128,128,0.3)"
```
[the remark this answers is not in the document]
Comment (nathancoliver) [reply dimes/common.py:193]: I moved them.

File: dimes/common.py [hunk 3]
```diff
@@ -182,6 +187,12 @@ def __init__(
...
+        self.RANGE_BUFFER = 0.1
```
[the remark this answers is not in the document]
Comment (nathancoliver) [reply dimes/common.py:194]: RANGE_BUFFER is now a module variable. Should I move this variable to within the function?

File: dimes/common.py [hunk 4]
```diff
@@ -182,6 +187,12 @@ def __init__(
             self.x_axis = x_axis
         self.subplots: List[Union[DimensionalSubplot, None]] = [None]
         self.is_finalized = False
+        self.GRID_LINE_WIDTH = 1.5
```
[the remark this answers is not in the document]
Comment (nathancoliver) [reply dimes/common.py:190]: GRID_LINE_WIDTH is currently being used in four locations. It is now a module variable if that is okay.

File: dimes/timeseries.py [hunk 5]
```diff
@@ -12,7 +13,8 @@ class TimeSeriesData(DisplayData):
 class TimeSeriesPlot(DimensionalPlot):
     """Time series plot."""
 
-    def __init__(self, time_values: list):
+    def __init__(self, time_values: list, plot_title: Union[str, None] = None):
         super().__init__(time_values)
         self.add_time_series = self.add_display_data
         self.time_values = self.x_axis.data_values
+        self.figure.layout["title"] = plot_title
```
[the remark this answers is not in the document]
Comment (nathancoliver) [reply dimes/timeseries.py:20]: I'll try and work this out

File: dimes/common.py [hunk 6]
```diff
@@ -57,6 +56,12 @@ class LinesOnly(LineProperties):
     marker_size: Union[int, None] = 0
 
 
+@dataclass
+class LegendProperties:
+    is_visible: bool = True
+    legend_group: Union[str, int, None] = None
```
[the remark this answers is not in the document]
Comment (nathancoliver) [reply dimes/common.py:62]: Okay

[the remark this answers is not in the document]
Comment (nathancoliver) [reply dimes/common.py:62]: It could be a year

File: dimes/common.py [hunk 7]
```diff
@@ -201,6 +212,17 @@ def add_display_data(
...
+    
+    def append_y_values_range(self, y_values):
+        self.y_range["min"].append(min(y_values))
+        self.y_range["max"].append(max(y_values))
+    
+    def get_axis_range(self, values):
+        min_value = min(values["min"])
+        max_value = max(values["max"])
+        range_values = max_value - min_value
+        return [min_value - range_values*self.RANGE_BUFFER, max_value + range_values*self.RANGE_BUFFER]
```
[the remark this answers is not in the document]
Comment (nathancoliver) [reply dimes/common.py:224]: In the acceptance criteria plots, I used a method to round up the max value to the nearest integer of five, and vise versa for the min value.  The tricky part will be in determining which nearest integer to round to, whether it's 1, 5, 100, etc.

File: dimes/common.py (hunk 1 above)
Comment (nathancoliver) [reply dimes/common.py:293]: Thanks

File: dimes/common.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (nathancoliver) [reply dimes/common.py:224]: Notes: set initial min and max values to positive inf and negative inf, respectively. Move set_axis_range to axis.

**Review by nealkruis — APPROVED** [review]

**Pull request merged**